### PR TITLE
feat: update expo example app to expo 47

### PR DESCRIPTION
**Pull request by santhoshvai**
## 🎯 Goal

Update the expo example app to expo 47 version

## 🛠 Implementation details

NA

## 🎨 UI Changes

<details>
<summary>iOS</summary>

<table>
    <thead>
        <tr>
            <td>Before</td>
            <td>After</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                
            </td>
            <td>
                
            </td>
        </tr>
    </tbody>
</table>
</details>

<details>
<summary>Android</summary>

<table>
    <thead>
        <tr>
            <td>Before</td>
            <td>After</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                
            </td>
            <td>
                
            </td>
        </tr>
    </tbody>
</table>
</details>

## ☑️ Checklist

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Documentation is updated
- [ ] New code is tested in main example apps, including all possible scenarios
  - [ ] SampleApp iOS and Android
  - [x] Expo iOS and Android

### CODE REVIEW

**Review by khushal87 — COMMENTED** [review]
Not able to run the changes on android:

```
BUILD FAILED in 1m 7s
10 actionable tasks: 4 executed, 6 up-to-date
Error: /Users/khushalagarwal/Stream/stream-chat-react-native/examples/ExpoMessaging/android/gradlew exited with non-zero code: 1
Error: /Users/khushalagarwal/Stream/stream-chat-react-native/examples/ExpoMessaging/android/gradlew exited with non-zero code: 1
    at ChildProcess.completionListener (/Users/khushalagarwal/Stream/stream-chat-react-native/examples/ExpoMessaging/node_modules/@expo/cli/node_modules/@expo/spawn-async/build/spawnAsync.js:52:23)
    at Object.onceWrapper (node:events:642:26)
    at ChildProcess.emit (node:events:527:28)
    at maybeClose (node:internal/child_process:1092:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5)
    ...
    at Object.spawnAsync [as default] (/Users/khushalagarwal/Stream/stream-chat-react-native/examples/ExpoMessaging/node_modules/@expo/cli/node_modules/@expo/spawn-async/build/spawnAsync.js:17:21)
    at spawnGradleAsync (/Users/khushalagarwal/Stream/stream-chat-react-native/examples/ExpoMessaging/node_modules/@expo/cli/build/src/start/platforms/android/gradle.js:72:46)
    at Object.assembleAsync (/Users/khushalagarwal/Stream/stream-chat-react-native/examples/ExpoMessaging/node_modules/@expo/cli/build/src/start/platforms/android/gradle.js:52:18)
    at runAndroidAsync (/Users/khushalagarwal/Stream/stream-chat-react-native/examples/ExpoMessaging/node_modules/@expo/cli/build/src/run/android/runAndroidAsync.js:31:24)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by santhoshvai** [comment]
@khushal87 delete your node modules of `stream-chat-react-native-core`.. another way is to update the stream-chat-react-native-core dev deps to this gesture handler version but that will break the other samples like this. Dev dependencies are ignored by the user, so this error wont happen to users of the SDK.

**Comment by santhoshvai** [comment]
aligned all gesture handler versions now @khushal87

File: package/package.json [hunk 1]
```diff
@@ -144,5 +144,9 @@
     "typescript": "4.4.3",
     "uuid": "^8.3.2",
     "webpack": "4.44.2"
+  },
+  "resolutions": {
+    "@types/react": "17.0.5",
+    "@types/react-native": "0.67.3"
```
Comment (santhoshvai) [inline package/package.json:150]: prevents types from other libs breaking our build. Happened with GH update.

**Comment by khushal87** [comment]
I will review it by today EOD @santhoshvai

**Comment by vanGalilea** [comment]
Closing this at the moment and we will revisit it later

**Pull request merged**